packages/jni-swig-stub/realm.i

```diff
@@ -57,6 +57,61 @@ std::string rlm_stdstr(realm_string_t val)
 }
 %}
 
+// Transforms a realm_index_range_t into a long array
+%typemap(in) (realm_index_range_t*, size_t) {
+    $2 = jenv->GetArrayLength($input);
+    $1 = new realm_index_range_t[$2];
+}
+
+%typemap(freearg) (realm_index_range_t*, size_t) {
+    for(int i = 0; i < $2; i++) {
+        jlongArray jRangeArray = (jlongArray) jenv->GetObjectArrayElement($input, i);
+        size_t *arg = (size_t *) 0 ;
+        jlong *jarr;
+
+        %#if defined(__ANDROID__)
```

Comment:
I guess this is a relatively tweaked typemap to optimize things that prevents using the normal typemap for LongArrays. 

The _straight forward_ (if there is such thing with Swig 🤪) is to configure Swig to generate classes for handling `realm_index_range_t` arrays just like we do for other use cases that requires transferring arrays of structs: https://github.com/realm/realm-kotlin/blob/master/packages/jni-swig-stub/realm.i#L202
Then you can allocate the array from Java as in https://github.com/realm/realm-kotlin/blob/master/packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt#L67 and access the structs as done in https://github.com/realm/realm-kotlin/blob/master/packages/cinterop/src/jvm/kotlin/io/realm/internal/interop/RealmInterop.kt#L94..L97. 

The specified typemap prevents crossing the JNI bridge a substantial number of times, but since the ideal way to do bulk transfers would be to write a custom general serializer, I don't know if we should optimize things like this now. If we go for this approach, then we should maybe just encapsulate the `#ifdef`'ed `SWIG_JavaArrayInLongLong` call in a custom method in `realm_api_helper` and use it here and in the other typemap tweaks where we do something similar.